File: android-custom-template-manager/Videosdk_android_java_hls_quickstart/app/src/main/java/live/videosdk/rtc/android/quickstart/MeetingActivity.java
package live.videosdk.rtc.android.quickstart;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;

import live.videosdk.rtc.android.Meeting;
import live.videosdk.rtc.android.VideoSDK;
import live.videosdk.rtc.android.listeners.MeetingEventListener;

public class MeetingActivity extends AppCompatActivity {

    private Meeting meeting;
    private String token;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_meeting);

        final String meetingId = getIntent().getStringExtra("meetingId");
        token = getIntent().getStringExtra("token");
        String mode = getIntent().getStringExtra("mode");
        String localParticipantName = "John Doe";
        boolean streamEnable = mode.equals("CONFERENCE");

        // initialize VideoSDK
        VideoSDK.initialize(getApplicationContext());

        // Configuration VideoSDK with Token
        VideoSDK.config(token);

        // Initialize VideoSDK Meeting
        meeting = VideoSDK.initMeeting(
                MeetingActivity.this, meetingId, localParticipantName,
                streamEnable, streamEnable,null, mode, true,null);

        // join Meeting
        meeting.join();

        // if mode is CONFERENCE than replace mainLayout with SpeakerFragment otherwise with ViewerFragment
        meeting.addEventListener(new MeetingEventListener() {
            @Override
            public void onMeetingJoined() {
                if (meeting != null) {
                    if (mode.equals("CONFERENCE")) {
                        meeting.getLocalParticipant().pin("SHARE_AND_CAM");
                        getSupportFragmentManager()
                                .beginTransaction()
                                .replace(R.id.mainLayout, new SpeakerFragment(), "MainFragment")
                                .commit();
                    } else if (mode.equals("VIEWER")) {
                        getSupportFragmentManager()
                                .beginTransaction()
                                .replace(R.id.mainLayout, new ViewerFragment(), "viewerFragment")
                                .commit();
                    }
                }
            }
        });
    }

    public Meeting getMeeting() {
        return meeting;
    }

    public String getToken()
    {
        return token;
    }

}

File: android-hls/Videosdk_android_kotlin_hls_quickstart/app/src/main/java/live/videosdk/rtc/android/quickstart/MeetingActivity.kt
package live.videosdk.rtc.android.quickstart

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import live.videosdk.rtc.android.Meeting
import live.videosdk.rtc.android.VideoSDK
import live.videosdk.rtc.android.listeners.MeetingEventListener

class MeetingActivity : AppCompatActivity() {
    var meeting: Meeting? = null
        private set

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_meeting)

        val meetingId = intent.getStringExtra("meetingId")
        val token = intent.getStringExtra("token")
        val mode = intent.getStringExtra("mode")
        val localParticipantName = "John Doe"
        val streamEnable = mode == "CONFERENCE"

        // initialize VideoSDK
        VideoSDK.initialize(applicationContext)

        // Configuration VideoSDK with Token
        VideoSDK.config(token)

        // Initialize VideoSDK Meeting
        meeting = VideoSDK.initMeeting(
            this@MeetingActivity, meetingId, localParticipantName,
            streamEnable, streamEnable, null, mode, true,null
        )

        // join Meeting
        meeting!!.join()

        // if mode is CONFERENCE than replace mainLayout with SpeakerFragment otherwise with ViewerFragment
        meeting!!.addEventListener(object : MeetingEventListener() {
            override fun onMeetingJoined() {
                if (meeting != null) {
                    if (mode == "CONFERENCE") {
                        meeting!!.localParticipant.pin("SHARE_AND_CAM")
                        supportFragmentManager
                            .beginTransaction()
                            .replace(R.id.mainLayout, SpeakerFragment(), "MainFragment")
                            .commit()
                    } else if (mode == "VIEWER") {
                        supportFragmentManager
                            .beginTransaction()
                            .replace(R.id.mainLayout, ViewerFragment(), "viewerFragment")
                            .commit()
                    }
                }
            }
        })
    }
}